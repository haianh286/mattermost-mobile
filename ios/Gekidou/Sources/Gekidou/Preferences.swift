//
//  File.swift
//  
//
//  Created by Elias Nahum on 26-06-22.
//

import Foundation

let appGroupId = Bundle.main.infoDictionary!["AppGroupIdentifier"] as? String

public class Preferences: NSObject {
    private let userDefauls = UserDefaults(suiteName: appGroupId)
    
    @objc public static let `default` = Preferences()
    
    override private init() {}
    
    public func object(forKey name: String) -> Any? {
        return userDefauls?.object(forKey: name)
    }
    
    public func removeObject(forKey name: String) {
        userDefauls?.removeObject(forKey: name)
    }
    
    public func set(_ value: Any?, forKey key: String) {
        userDefauls?.set(value, forKey: key)
    }
}
