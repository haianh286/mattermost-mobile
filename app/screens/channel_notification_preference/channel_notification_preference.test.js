// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {ViewTypes} from '@constants';
import Preferences from '@mm-redux/constants/preferences';
import SectionItem from '@screens/settings/section_item';
import {shallowWithIntlMessages} from '@test/intl-test-helper';

import ChannelNotificationPreference from './channel_notification_preference';

function makeProps(pushNotificationLevel) {
    return {
        actions: {
            updateChannelNotifyProps: jest.fn(),
        },
        channelId: 'channel_id',
        globalNotifyProps: {
            push: 'mention',
        },
        userId: 'user_id',
        notifyProps: {
            push: pushNotificationLevel,
        },
        theme: Preferences.THEMES.default,
    };
}

function checkNotificationSelected(pushNotificationLevel, trueIdx) {
    const wrapper = shallowWithIntlMessages(
        <ChannelNotificationPreference
            {...makeProps(pushNotificationLevel)}
        />,
    );

    const sectionItems = wrapper.find(SectionItem);

    expect(sectionItems.exists()).toBe(true);
    expect(sectionItems.length).toBe(4);

    sectionItems.forEach((sectionItem, idx) => {
        expect(sectionItem.prop('selected')).toBe(idx === trueIdx);
    });
}

describe('ChannelNotificationPreference', () => {
    test('should match snapshot', () => {
        const baseProps = makeProps('default');
        const wrapper = shallowWithIntlMessages(
            <ChannelNotificationPreference {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should have correct setting selected', () => {
        checkNotificationSelected(null, 0);
        checkNotificationSelected(ViewTypes.NotificationLevels.DEFAULT, 0);
        checkNotificationSelected(ViewTypes.NotificationLevels.ALL, 1);
        checkNotificationSelected(ViewTypes.NotificationLevels.MENTION, 2);
        checkNotificationSelected(ViewTypes.NotificationLevels.NONE, 3);
    });

    test('should save on click', () => {
        const props = makeProps('default');
        const wrapper = shallowWithIntlMessages(
            <ChannelNotificationPreference {...props}/>,
        );

        // click on 'Never' -- last item
        wrapper.find(SectionItem).at(3).dive().simulate('press');

        expect(props.actions.updateChannelNotifyProps).toHaveBeenCalledTimes(1);
        expect(props.actions.updateChannelNotifyProps).toBeCalledWith(
            props.userId,
            props.channelId,
            {push: ViewTypes.NotificationLevels.NONE},
        );
    });
});
