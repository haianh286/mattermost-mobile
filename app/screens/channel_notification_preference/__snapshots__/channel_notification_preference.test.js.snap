// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelNotificationPreference should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(61,60,64,0.03)",
      "flex": 1,
    }
  }
  testID="channel_notification_preference.screen"
>
  <Connect(StatusBar) />
  <RNCSafeAreaView
    edges={
      Array [
        "left",
        "right",
      ]
    }
  >
    <InjectIntl(FormattedText)
      defaultMessage="Send Notifications"
      id="channel_notifications.preference.header"
      style={
        Object {
          "color": "rgba(61,60,64,0.56)",
          "fontSize": 13,
          "marginTop": 10,
          "padding": 16,
          "textTransform": "uppercase",
        }
      }
    />
  </RNCSafeAreaView>
  <ScrollView
    alwaysBounceVertical={false}
    contentContainerStyle={
      Object {
        "backgroundColor": "rgba(61,60,64,0.03)",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,1)",
        }
      }
    >
      <RNCSafeAreaView
        edges={
          Array [
            "left",
            "right",
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "backgroundColor": "rgba(61,60,64,0.1)",
                "height": 1,
                "marginLeft": 16,
                "width": "100%",
              }
            }
          />
          <sectionItem
            action={[Function]}
            actionType="select"
            actionValue="default"
            label={
              <InjectIntl(FormattedText)
                defaultMessage="Global default ({notifyLevel})"
                id="channel_notifications.preference.global_default"
                values={
                  Object {
                    "notifyLevel": "Mentions",
                  }
                }
              />
            }
            selected={true}
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBg": "#ffffff",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
          />
        </View>
        <View>
          <View
            style={
              Object {
                "backgroundColor": "rgba(61,60,64,0.1)",
                "height": 1,
                "marginLeft": 16,
                "width": "100%",
              }
            }
          />
          <sectionItem
            action={[Function]}
            actionType="select"
            actionValue="all"
            label={
              <InjectIntl(FormattedText)
                defaultMessage="For all activity"
                id="channel_notifications.preference.all_activity"
              />
            }
            selected={false}
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBg": "#ffffff",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
          />
        </View>
        <View>
          <View
            style={
              Object {
                "backgroundColor": "rgba(61,60,64,0.1)",
                "height": 1,
                "marginLeft": 16,
                "width": "100%",
              }
            }
          />
          <sectionItem
            action={[Function]}
            actionType="select"
            actionValue="mention"
            label={
              <InjectIntl(FormattedText)
                defaultMessage="Only mentions and direct messages"
                id="channel_notifications.preference.only_mentions"
              />
            }
            selected={false}
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBg": "#ffffff",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
          />
        </View>
        <View>
          <View
            style={
              Object {
                "backgroundColor": "rgba(61,60,64,0.1)",
                "height": 1,
                "marginLeft": 16,
                "width": "100%",
              }
            }
          />
          <sectionItem
            action={[Function]}
            actionType="select"
            actionValue="none"
            label={
              <InjectIntl(FormattedText)
                defaultMessage="Never"
                id="channel_notifications.preference.never"
              />
            }
            selected={false}
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBg": "#ffffff",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
          />
        </View>
      </RNCSafeAreaView>
      <View
        style={
          Object {
            "backgroundColor": "rgba(61,60,64,0.1)",
            "height": 1,
            "marginLeft": 16,
            "width": "100%",
          }
        }
      />
    </View>
  </ScrollView>
</View>
`;
